test/cli/formatters/sarif.spec.js

```diff
@@ -0,0 +1,70 @@
+const ChildProcess = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const os = require('os')
+
+describe('CLI', () => {
+  describe('Formatter: sarif', () => {
+    it('should have stdout output with formatter sarif', (done) => {
+      const expectedFileContent = fs
+        .readFileSync(path.resolve(__dirname, 'sarif.sarif'), 'utf8')
+        .replace(
+          /\{\{path\}\}/g,
+          path.resolve(__dirname, 'example.html').replace(/\\/g, '\\\\')
+        )
+
+      const expected = JSON.parse(expectedFileContent)
+
+      var child = ChildProcess.spawn('node', [
+        path.resolve(__dirname, '../../../bin/htmlhint'),
+        path.resolve(__dirname, 'example.html'),
+        '--format',
+        'sarif',
+      ])
+
+      child.stdout.on('data', function (stdout) {
+        expect(stdout).not.toBe('')
+
+        if (os.platform() !== 'darwin') {
```

Comment:
I have problem with this test running in macOS,
at first I was using the exact same way `ChildProcess.exec`, copied from `json.spec.js`. It seems macOS will truncate because sarif is too long.
So I tried to change to `ChildProcess.spawn`, seems like still have same issue. Not familiar with NodeJS, Let me know.
This looks like related:
[https://github.com/nodejs/node/issues/19218](https://github.com/nodejs/node/issues/19218)

Reply:
Sorry, maybe I didn't land it, but could this use the Jest snapshoting to work around it?

Reply:
Thanks for looking,
the issue was we are comparing the output against a pre-saved file, 
in macOS only, the stdout seems have limit, and will lose part of the json,
the error was: 
`SyntaxError: Unexpected end of JSON input` at `at JSON.parse (<anonymous>)`
`> 34 |           const jsonStdout = JSON.parse(stdout)`

the tests for Linux and Windows does not have this issue and pass.

Reply:
In the above thread in nodejs repo, I can see we had a same issue before. https://github.com/htmlhint/HTMLHint/pull/442
Our fix was trim down the size of the test. I can try to do the same to fix this as well so the test can be ran in macOS.